packages/ui/src/pages/Metadata/MetadataPage.tsx

```diff
@@ -0,0 +1,51 @@
+import { Title } from '@patternfly/react-core';
+import { FunctionComponent, useContext } from 'react';
+import { MetadataEditor } from '../../components/MetadataEditor';
+import { useSchemasStore } from '../../store';
+import { EntitiesContext } from '../../providers/entities.provider';
+import { ObjectMeta } from '@kaoto-next/camel-catalog/types';
+import { EntityType } from '../../models/camel/entities';
+import { MetadataEntity } from '../../models/visualization/metadata';
+
+export const MetadataPage: FunctionComponent = () => {
+  const schemaMap = useSchemasStore((state) => state.schemas);
+  const entitiesContext = useContext(EntitiesContext);
+  const entities = entitiesContext?.entities ?? [];
+
+  function findMetadata() {
```

Comment:
[medium]: for functions inside react components, it's recommended to use arrow functions through the `useCallback` hook, otherwise these functions will be recreated on every render.

Reply:
done :+1: